spyder/widgets/sourcecode/tests/test_folding.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Tests for the autoindent features
+"""
+
+# Third party imports
+from qtpy.QtGui import QTextCursor
+import pytest
+
+# Local imports
+from spyder.utils.qthelpers import qapplication
+from spyder.widgets.sourcecode.codeeditor import CodeEditor
+from spyder.widgets.sourcecode.api.folding import print_tree
+
+
+# --- Fixtures
+# -----------------------------------------------------------------------------
+@pytest.fixture()
+def get_fold_levels():
+    """setup editor and return fold levels."""
+
+    def get_fold_levels():
```

Comment:
The fixture will return a function, It will allow the fixture to receive parameters, maybe the text for the editor, or another FoldDetector that will be useful for other tests.

I found this trick in stackoverflow, although reading a little more is better to use [indirect parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments) I'll change it